dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/upgrade/DolphinSchedulerManager.java

```diff
@@ -78,6 +78,13 @@ public void initDolphinSchedulerSchema() {
 
         logger.info("Start initializing the DolphinScheduler manager table structure");
         upgradeDao.initSchema();
+        //create directories in the sql/create to store the multiple upgrade mirrors
+        try {
+            UpgradeDao.createMirrorDirs();
```

Comment:
Hi,Why create mirror dirs here?And what's the different from the ddl under sql/create and sql/upgrade?

Reply:
I think you've done much better this time than you did last time.Now you only need design the ddl and dml under sql/create,where the file of mirror ddls and dmls come from?You can see the example of existed files of sql/createm.

Reply:
>
>
> Hi,Why create mirror dirs here?And what's the different from the ddl under sql/create and sql/upgrade?

Well, I wanted the mirror directories to be created once the DolphinScheduler is initialized, I thought that was the right way, is there any better way, am eager for suggestions, thanks

>
>
> I think you've done much better this time than you did last time.Now you only need design the ddl and dml under sql/create,where the file of mirror ddls and dmls come from?You can see the example of existed files of sql/createm.

Thank you very much, but the big question for me is; all this creating new ddl and dml is new to me, so how can I get them from iterating sql/create?! Thanks